zerver/views/public_export.py

```diff
@@ -0,0 +1,35 @@
+import datetime
+
+from django.utils.timezone import now as timezone_now
+from django.utils.translation import ugettext as _
+from django.http import HttpResponse
+
+from zerver.models import RealmAuditLog, Realm
+from zerver.lib.queue import queue_json_publish
+from zerver.lib.response import json_error, json_success
+
+def public_only_realm_export(realm: Realm) -> HttpResponse:
+    event_type = RealmAuditLog.REALM_EXPORT_TRIGGERED
+    event_time = timezone_now()
+    # Check for event times occurring in the past week
+    event_time_delta = event_time - datetime.timedelta(days=7)
+
+    # Basic rate limiting for public-only realm exports
+    # We check if the nth from the last record's event time occurred within the time delta
+    # If true, the request to export the realm exceeds the weekly limit
+    limit_check = RealmAuditLog.objects.filter(realm=realm,
+                                               event_type=event_type).order_by("-event_time")[:5].last()
```

Comment:
Checking the last 5 RealmAuditLog events doesn't make sense -- it'll create potential security issues.  We should just filter the database query with e.g. `event_time__gte=event_time_delta`).